libs/ledger-live-common/src/families/ethereum/synchronisation.ts

```diff
@@ -54,31 +55,55 @@ export const getAccountShape: GetAccountShape = async (
   const blockHashExistsOnChain = await api
     .getBlockByHash(mostRecentStableOperation?.blockHash)
     .then(Boolean);
+
   const syncHash =
     JSON.stringify(blacklistedTokenIds || []) +
     "_" +
     listTokensForCryptoCurrency(currency, {
       withDelisted: true,
     }).length;
+
   const outdatedSyncHash = initialAccount?.syncHash !== syncHash;
+  const allOperationsLoaded = initialAccount
+    ? areAllOperationsLoaded(initialAccount)
+    : false;
+
   const pullFromBlockHeight =
     initialAccount &&
-    areAllOperationsLoaded(initialAccount) &&
+    allOperationsLoaded &&
     mostRecentStableOperation &&
     blockHashExistsOnChain &&
     !outdatedSyncHash
       ? mostRecentStableOperation.blockHeight
       : undefined;
+
   const txsP = fetchAllTransactions(api, address, pullFromBlockHeight);
   const [txs, currentBlock, balance] = await Promise.all([
     txsP,
     currentBlockP,
     balanceP,
   ]);
-  const blockHeight = currentBlock.height.toNumber();
+
+  const blockHeight = currentBlock?.height.toNumber();
+
+  log("Ethereum", "fetchAllTransactions", txs);
+  log("Ethereum", "pullFromBlockHeight conditions", {
+    initialAccount,
+    allOperationsLoaded,
+    mostRecentStableOperation,
+    blockHashExistsOnChain,
+    outdatedSyncHash,
+  });
 
   if (!pullFromBlockHeight && txs.length === 0) {
     log("ethereum", "no ops on " + address);
+    log(
+      "ethereum",
+      `Current Block { hash: ${currentBlock?.hash}, height: ${
+        currentBlock?.height
+      }, txs: ${currentBlock?.txs.join(" ===== ")} }`
```

Comment:
Same here, logging all of the currentBlock transactions might be an issue. Also I'm not sure we're using those in the context of the synchronisation.